client/components/Checkout.js

```diff
@@ -0,0 +1,447 @@
+/* eslint-disable indent */
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import {
+  fetchOrder as fetchOrderThunk,
+  fetchOrderItems,
+  updateOrderThunk,
+  updateUser,
+} from '../store';
+import { isLoggedIn } from './helperFunctions';
+
+class Checkout extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      shippingAddress: '',
+      shippingCity: '',
+      shippingState: '',
+      shippingZip: '',
+      billingFirstName: '',
+      billingLastName: '',
+      billingAddress: '',
+      billingCity: '',
+      billingState: '',
+      billingZip: '',
+      ccNumber: '',
+      ccv: '',
+      ccExpDateMonth: 1,
+      ccExpDateYear: 2019,
+      sameAddress: false,
+      submitted: false,
+    };
+  }
+
+  componentDidMount = () => {
+    const { fetchOrder, user, order } = this.props;
+    fetchOrder(user.id);
+    fetchOrderItems(order.id);
+  };
+
+  handleChange = ev => {
+    ev.preventDefault();
+
+    const {
+      firstName,
+      lastName,
+      shippingAddress,
+      shippingCity,
+      shippingState,
+      shippingZip,
+      sameAddress,
+    } = this.state;
+
+    this.setState({ [`${ev.target.name}`]: ev.target.value });
+
+    if (sameAddress) {
+      this.setState({
+        billingFirstName: firstName,
+        billingLastName: lastName,
+        billingAddress: shippingAddress,
+        billingCity: shippingCity,
+        billingState: shippingState,
+        billingZip: shippingZip,
+      });
+    }
+  };
+
+  handleSubmit = ev => {
+    const {
+      firstName,
+      lastName,
+      shippingAddress,
+      shippingCity,
+      shippingState,
+      shippingZip,
+      billingFirstName,
+      billingLastName,
+      billingAddress,
+      billingCity,
+      billingState,
+      billingZip,
+      ccNumber,
+      ccv,
+      ccExpDateMonth,
+      ccExpDateYear,
+      email,
+      phone,
+    } = this.state;
+
+    const { updateOrder, updateUserThunk, order, user } = this.props;
+
+    ev.preventDefault();
+
+    updateOrder({
+      ...order,
+      status: 'processing',
+      shippingAddress,
+      shippingCity,
+      shippingState,
+      shippingZip: parseInt(shippingZip),
+      date: new Date(),
+    });
+
+    updateUserThunk({
+      ...user,
+      firstName,
+      lastName,
+      billingFirstName,
+      billingLastName,
+      billingAddress,
+      billingCity,
+      billingState,
+      billingZip: parseInt(billingZip),
+      email,
+      phone: parseInt(phone),
+      ccNumber: parseInt(ccNumber),
```

Comment:
We never discussed how to handle payments, but probably should have.  We can not request and store cc info.  There are just too many security issues related to this.  This is why we will be using a third party payment solution like Square (or Stripe if we can't get Square to work) where all of the payment info gets collected, stored and processed on their servers.

Leave all of this logic here for now and then we can deal with it when integrating a third-party payment solution.  Still it was a good exercise in thinking about what info needs to be collected.

It should be treated as a **string** of numbers, not one number.
```suggestion
      ccNumber: ccNumber,
```